src/utils/datetime.ts

```diff
@@ -0,0 +1,370 @@
+ * @returns The value as a string in the current locale.
+ * @throws {Error} If the value parameter is null.
+ */
+export function toLocaleString(value: Date | number | null | undefined, options?: Intl.NumberFormatOptions | Intl.DateTimeFormatOptions): string {
+    if (!value) {
+        throw new Error('value cannot be null');
+    }
+
+    options = options || {};
+
+    if (toLocaleTimeStringSupportsLocales) {
+        const currentLocale = globalize.getCurrentDateTimeLocale();
+
+        if (currentLocale) {
+            return value.toLocaleString(currentLocale, options);
+        }
+    }
+
+    return value.toLocaleString();
+}
+
+/**
+ * Converts a date to a date string using the current locale.
+ * @param date - The date to convert to a string.
+ * @param options - An optional object containing formatting options.
+ * @returns The date part of the date as a string in the current locale.
+ * @throws {Error} If the date parameter is null.
+ */
+export function toLocaleDateString(date: DateOrNull, options?: Intl.DateTimeFormatOptions): string {
+    if (!date) {
+        throw new Error('date cannot be null');
+    }
+
+    options = options || {};
+
+    if (toLocaleTimeStringSupportsLocales) {
+        const currentLocale = globalize.getCurrentDateTimeLocale();
+
+        if (currentLocale) {
+            return date.toLocaleDateString(currentLocale, options);
+        }
+    }
+
+    // This is essentially a hard-coded polyfill
+    const optionList = getOptionList(options);
+    if (optionList.length === 1 && optionList[0].name === 'weekday') {
+        const weekday = [];
+        weekday[0] = 'Sun';
+        weekday[1] = 'Mon';
+        weekday[2] = 'Tue';
+        weekday[3] = 'Wed';
+        weekday[4] = 'Thu';
+        weekday[5] = 'Fri';
+        weekday[6] = 'Sat';
+        return weekday[date.getDay()];
+    }
+
+    return date.toLocaleDateString();
+}
+
+/**
+ * Converts a date to a time string using the current locale.
+ * @param date - The date to convert to a string.
+ * @param options - An optional object containing formatting options.
+ * @returns The time part of the date as a string in the current locale.
+ * @throws {Error} If the value parameter is null.
+ */
+export function toLocaleTimeString(date: DateOrNull, options?: Intl.DateTimeFormatOptions): string {
+    if (!date) {
+        throw new Error('date cannot be null');
+    }
+
+    options = options || {};
+
+    if (toLocaleTimeStringSupportsLocales) {
+        const currentLocale = globalize.getCurrentDateTimeLocale();
+
+        if (currentLocale) {
+            return date.toLocaleTimeString(currentLocale, options);
+        }
+    }
+
+    return date.toLocaleTimeString();
+}
+
+/**
+ * Checks if the given value is a string.
+ * @param value - The value to check.
+ * @returns True if the value is a string, false otherwise.
+ */
+function isString(value: Date | string): value is string {
+    return (typeof value).toString().toLowerCase() === 'string';
```

Comment:
Why would we not just use `typeof value === 'string'`? This seems like a legacy code smell.